src/mcts/search.cc

```diff
@@ -556,18 +567,38 @@ void SearchWorker::GatherMinibatch() {
       if (++collisions_found > search_->kAllowedNodeCollisions) return;
       continue;
     }
-    ++nodes_found;
+    ++minibatch_size;
+
     // If node is already known as terminal (win/loss/draw according to rules
     // of the game), it means that we already visited this node before.
-    if (node->IsTerminal()) continue;
+    if (!node->IsTerminal()) {
+      // Node was never visited, extend it.
+      ExtendNode(node);
 
-    // Node was never visited, extend it.
-    ExtendNode(node);
+      // Only send non-terminal nodes to neural network
+      if (!node->IsTerminal()) {
+        minibatch_.back().nn_queried = true;
+        minibatch_.back().is_cache_hit = AddNodeToComputation(node, true);
```

Comment:
You already have the `auto& picked_node = minibatch_.back()` local variable, may as well use it here

Reply:
Done.